### PR TITLE
Removes redunant variables from code blocks in add-w-and-b-to-your-code.md

### PR DESCRIPTION
**Pull request by SinaMostafanejad**
{Description of the change}

This PR:
- removes the unused **run** variables from the code blocks, and
- fixes a minor typo ("," -> ":") in one of the code blocks

### CODE REVIEW

File: content/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 1]
```diff
@@ -113,7 +113,7 @@ def evaluate_one_epoch(epoch):
 
 
 def main():
-    run = wandb.init()
+    wandb.init()
```
Comment (mdlinville) [inline content/guides/models/sweeps/add-w-and-b-to-your-code.md]: I believe that this change and the one on line 203 are in conflict with https://github.com/wandb/docs/pull/1146, which recently merged. Can you take a look and let me know what you think?

**Review by mdlinville — CHANGES_REQUESTED** [review]

File: content/guides/models/sweeps/add-w-and-b-to-your-code.md (hunk 1 above)
Comment (SinaMostafanejad) [reply content/guides/models/sweeps/add-w-and-b-to-your-code.md]: Agreed. We might as well either use the recommended context manager expression or use **run.finish()** to ensure the completion of the run as described here (https://docs.wandb.ai/guides/runs/#initialize-a-run)

Comment (mdlinville) [reply content/guides/models/sweeps/add-w-and-b-to-your-code.md]: I checked with @timoffex who did the recent edits, and lines 120-122 here should be updated according to this pattern:

```
lr = run.config["lr"]
bs = run.config["batch_size"]
epochs = run.config["epochs"]
```

File: content/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 2]
```diff
@@ -200,7 +200,7 @@ def main():
     with open("./config.yaml") as file:
         config = yaml.load(file, Loader=yaml.FullLoader)
 
-    run = wandb.init(config=config)
+    wandb.init(config=config)
 
     # Note that we define values from `wandb.config`
     # instead of  defining hard values
```
Comment (mdlinville) [inline content/guides/models/sweeps/add-w-and-b-to-your-code.md:218]: These should be updated in a similar way

**Review by mdlinville — CHANGES_REQUESTED** [review]

File: content/guides/models/sweeps/add-w-and-b-to-your-code.md (hunk 2 above)
Comment (SinaMostafanejad) [reply content/guides/models/sweeps/add-w-and-b-to-your-code.md:218]: Yes. I am editing the code blocks the same way you just mentioned and they should be ready soon. I will try to run them locally and if they work without issues, I will commit them for your review.

Comment (SinaMostafanejad) [reply content/guides/models/sweeps/add-w-and-b-to-your-code.md:218]: I have edited the code block and the Python SDK panel guidelines. You can check the run here: https://wandb.ai/molssiai-tests/my-first-sweep/sweeps/paiae01m/workspace?nw=nwusersinamostafanejad

Comment (SinaMostafanejad) [reply content/guides/models/sweeps/add-w-and-b-to-your-code.md:218]: If this is what we want, I can edit the YAML file section as well. What makes me a little confused is that in Python SDK panel code, we use the config dictionary as an argument to sweep function but in the YAML code, we pass it to the `wandb.init()` as we do the sweep through the CLI and using the YAML file directly
`wandb sweep --project sweep-demo-cli config.yaml`

Two questions: 
- is it necessary to pass the config object to `wandb.init()` as we did in the YAML code panel?
`run = wandb.init(config=config)`
- Why did/do not we pass the config  object to `wandb.init()` in the Python SDK code panel?

Is it going to be redundant/incorrect to pass the sweep config dictionary to `wandb.init()`? when the `wandb.sweep()` will also consume it in the Python code? I think I can get my answers by looking at the code between the sweep and run classes but I'd appreciate any insights.

Comment (ngrayluna) [reply content/guides/models/sweeps/add-w-and-b-to-your-code.md:218]: The workflow of creating a sweep agent and starting the sweep itself differs if you use the CLI vs the Python SDK. Here are the differences:

### CLI 
* Sweep config defined in a YAML file
* Do not need to wrap the logic of your sweep in a function and pass that to the sweep agent. 
* Initialize (get sweep ID ) with:
   ```bash
   wandb sweep --project sweep-demo-cli config.yaml
   ```
* Start the sweep with:
   ```bash 
   wandb agent --count $NUM your-entity/sweep-demo-cli/sweepID
  ```

### Python script:
* Sweep config defined in a dict 
   ```python
   sweep_configuration = {
    "method": "random",
    "name": "sweep",
    "metric": {"goal": "maximize", "name": "val_acc"}}
   sweep_id = wandb.sweep(sweep=sweep_configuration, project="my-first-sweep")
   ```
* Define a function that contains your training script logic and pass that to sweep agent
   ```python
   def objective(config):
       score = config.x**3 + config.y
       return score

   def main():
       wandb.init(project="my-first-sweep")
       score = objective(wandb.config)
       wandb.log({"score": score})

   wandb.agent(sweep_id, function=main, count=10)
   ```

Comment (SinaMostafanejad) [reply content/guides/models/sweeps/add-w-and-b-to-your-code.md:218]: Thanks. Sure, but on the Python side, I think it is programmatically valid (no errors raised) to pass a dictionary to either initialize a Run object (https://github.com/wandb/wandb/blob/main/wandb/apis/public/runs.py#L276-L300) or a Sweep object (https://github.com/wandb/wandb/blob/main/wandb/apis/public/sweeps.py#L30-L47).
Note the sweep config dictionary can have ranges (max and min keys with values) that may not be valid entries for single Runs.

My guess is it is not going to be a problem even if we pass the same dictionary to both constructors because at the end, Sweep feeds the Run through attributes apparently and the invalid run entries in the config dictionary for Run (such as ranges with min and max keys) will be correctly handled by Sweep and fixed in the individual Runs.

Comment (ngrayluna) [reply content/guides/models/sweeps/add-w-and-b-to-your-code.md:218]: Correct. This is another option. 

For readability, some users prefer the delineation of using config dicts passed to `wandb.init()` for non hyperparameter sweep info. (as opposed to combining them).

Comment (SinaMostafanejad) [reply content/guides/models/sweeps/add-w-and-b-to-your-code.md:218]: Perfect. @mdlinville I think this part can go forward.

File: content/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 3]
```diff
@@ -114,7 +114,7 @@ import random
 
 # Define training function that takes in hyperparameter
 # values from `wandb.config` and uses them to train a
-# model and return metric
+# model and return the metric(s)
```
Comment (mdlinville) [inline content/guides/models/sweeps/add-w-and-b-to-your-code.md]:
```suggestion
# model and return the metrics
```
In general, if it could be singular or plural, just use the plural :)

File: content/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 4]
```diff
@@ -139,20 +139,21 @@ sweep_configuration = {
...
 def main():
-    # Use with context to automatically end the run
-    # equivalent to using `run.finish()` at the end of each run
+    # Use the `with` context manager statement to automatically end the run.
+    # This is equivalent to using `run.finish()` at the end of each run
     with wandb.init(project=project) as run:
 
-        # note that we define values from `wandb.config`
-        # instead of defining hard values
+        # Note that we fetch the hyperparameter values from `wandb.config`
```
Comment (mdlinville) [inline content/guides/models/sweeps/add-w-and-b-to-your-code.md]:
```suggestion
        # This code fetches the hyperparameter values from `wandb.config`
```

**Review by mdlinville — APPROVED** [review]
LGTM, just a few tiny nits, but I will go ahead and approve so that you can enable auto-merge. Thanks for your patience!

**Comment by SinaMostafanejad** [comment]
Thank you @mdlinville for the help. The CLI panel and the rest of the page also need a heavy revision too. That maybe better saved for another time, a separate PR and a cleaner chain of discussion.

File: content/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 5]
```diff
@@ -8,7 +8,9 @@ title: Add W&B (wandb) to your code
 weight: 2
 ---
 
-There are numerous ways to add the W&B Python SDK to your script or Jupyter Notebook. Outlined below is a "best practice" example of how to integrate the W&B Python SDK into your own code.
+There are numerous ways to add the W&B Python SDK to your script or Jupyter
+Notebook. Outlined below is a "best practice" example of how to integrate the
+W&B Python SDK into your own code.
```
Comment (mdlinville) [inline content/guides/models/sweeps/add-w-and-b-to-your-code.md]:
```suggestion
There are numerous ways to add the W&B Python SDK to your script or notebook. This section provides a "best practice" example that shows how to integrate the W&B Python SDK into your own code.

```

File: content/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 6]
```diff
@@ -157,7 +178,8 @@ parameters:
 
 For more information on how to create a W&B Sweep configuration, see [Define sweep configuration]({{< relref "/guides/models/sweeps/define-sweep-configuration/" >}}).
 
-Note that you must provide the name of your Python script for the `program` key in your YAML file.
+Note that you must provide the name of your Python script for the `program` key
```
Comment (mdlinville) [inline content/guides/models/sweeps/add-w-and-b-to-your-code.md]:
```suggestion
You must provide the name of your Python script for the `program` key
```

File: content/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 7]
```diff
@@ -219,36 +241,43 @@ def main():
 main()
 ```
 
-Navigate to your CLI. Within your CLI, set a maximum number of runs the sweep agent should try. This is step optional. In the following example we set the maximum number to five.
+Navigate to your CLI. Within your CLI, set a maximum number of runs the sweep
```
Comment (mdlinville) [inline content/guides/models/sweeps/add-w-and-b-to-your-code.md]:
```suggestion
In your CLI, set a maximum number of runs for the sweep
```

File: content/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 8]
```diff
@@ -219,36 +241,43 @@ def main():
...
+agent should try. This is step optional. In the following example we set the
```
Comment (mdlinville) [inline content/guides/models/sweeps/add-w-and-b-to-your-code.md]:
```suggestion
agent to try. This is optional. This example we set the
```

File: content/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 9]
```diff
@@ -219,36 +241,43 @@ def main():
...
+maximum number to five.
```
Comment (mdlinville) [inline content/guides/models/sweeps/add-w-and-b-to-your-code.md]:
```suggestion
maximum number to 5.
```

File: content/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 10]
```diff
@@ -219,36 +241,43 @@ def main():
...
 
 ```bash
 NUM=5
 ```
 
-Next, initialize the sweep with the [`wandb sweep`]({{< relref "/ref/cli/wandb-sweep.md" >}}) command. Provide the name of the YAML file. Optionally provide the name of the project for the project flag (`--project`):
+Next, initialize the sweep with the [`wandb
+sweep`]({{< relref "/ref/cli/wandb-sweep.md" >}}) command. Provide the name of
+the YAML file. Optionally provide the name of the project for the project flag
+(`--project`):
```
Comment (mdlinville) [inline content/guides/models/sweeps/add-w-and-b-to-your-code.md]:
```suggestion
Next, initialize the sweep with the [`wandb sweep`]({{< relref "/ref/cli/wandb-sweep.md" >}}) command. Provide the name of the YAML file. Optionally provide the name of the project for the project flag (`--project`):
```
For the future, Markdown diffs are much easier to follow and leave feedback on if you do not hard wrap the lines. If you must hard wrap, it's best to avoid splitting up the Markdown link text.

File: content/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 11]
```diff
@@ -219,36 +241,43 @@ def main():
...
+the sweep job with the [`wandb agent`]({{< relref "/ref/cli/wandb-agent.md" >}})
+command:
 
 ```bash
 wandb agent --count $NUM your-entity/sweep-demo-cli/sweepID
 ```
 
-For more information on how to start sweep jobs, see [Start sweep jobs]({{< relref "./start-sweep-agents.md" >}}).
+For more information on how to start sweep jobs, see [Start sweep
+jobs]({{< relref "./start-sweep-agents.md" >}}).
```
Comment (mdlinville) [inline content/guides/models/sweeps/add-w-and-b-to-your-code.md]:
```suggestion
For more information on how to start sweep jobs, see [Start sweep jobs]({{< relref "./start-sweep-agents.md" >}}).
```

File: content/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 12]
```diff
@@ -219,36 +241,43 @@ def main():
...
+{{% /tab %}} {{< /tabpane >}}
 
+## Consideration when logging metrics
 
-
-## Consideration when logging metrics 
-
-Ensure to log the metric you specify in your sweep configuration explicitly to W&B. Do not log metrics for your sweep inside of a sub-directory. 
+Ensure to log the metric you specify in your sweep configuration explicitly to
+W&B. Do not log metrics for your sweep inside of a sub-directory.
```
Comment (mdlinville) [inline content/guides/models/sweeps/add-w-and-b-to-your-code.md]:
```suggestion
Be sure to log the sweep's metric to W&B explicitly. Do not log metrics for your sweep inside a subdirectory.
```

**Review by mdlinville — COMMENTED** [review]
Apologies, I didn't realize that we didn't get this PR merged yet. Here is some final feedback.

**Comment by SinaMostafanejad** [comment]
> Apologies, I didn't realize that we didn't get this PR merged yet. Here is some final feedback.

Thank you @mdlinville for the suggestions.

File: content/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 13]
```diff
@@ -269,15 +264,13 @@ command:
 wandb agent --count $NUM your-entity/sweep-demo-cli/sweepID
 ```
 
-For more information on how to start sweep jobs, see [Start sweep
-jobs]({{< relref "./start-sweep-agents.md" >}}).
+For more information on how to start sweep jobs, see [Start sweep jobs]({{< relref "./start-sweep-agents.md" >}}).
```
Comment (mdlinville) [inline content/guides/models/sweeps/add-w-and-b-to-your-code.md]:
```suggestion
For more information, see [Start sweep jobs]({{< relref "./start-sweep-agents.md" >}}).
```

**Review by mdlinville — APPROVED** [review]
